File: php/src/Amen/Resources/Webhooks.php
<?php
declare(strict_types=1);
namespace Amen\Resources;
use Amen\AmenClient;
final class Webhooks
{
    public function __construct(private AmenClient $c) {}
    public function list(): array { return $this->c->request('GET', '/web-hooks/'); }
    /** `secret_key` in the response is shown ONLY now — store it in a secret manager immediately. */
    public function create(string $url): array { return $this->c->request('POST', '/web-hooks/', ['json' => ['url' => $url]]); }
    public function delete(string $id): void { $this->c->request('DELETE', "/web-hooks/$id"); }
}

File: kotlin/amen-client/src/main/kotlin/sa/amnn/kit/resources/Webhooks.kt
package sa.amnn.kit.resources

import kotlinx.serialization.builtins.ListSerializer
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import sa.amnn.kit.AmenClient
import sa.amnn.kit.Webhook

class Webhooks(private val c: AmenClient) {
    suspend fun list() = c.request("GET", "/web-hooks/", ListSerializer(Webhook.serializer())) ?: emptyList()
    /** secretKey in the response is shown ONLY now — store it in a secret manager immediately. */
    suspend fun create(url: String) = c.request("POST", "/web-hooks/", Webhook.serializer(), body = buildJsonObject { put("url", url) })!!
    suspend fun delete(id: String) { c.request("DELETE", "/web-hooks/$id", JsonElement.serializer()) }
}
